Step towards removing RenderSimple and its derivatives (Part 1)

reaperrr opened the pull request:
This PR gets some of the easy stuff out of the way to reduce the diff and improve reviewability of the follow-ups.

reaperrr committed:
Fix WithSpriteBody PlayCustomAnimationRepeating

Lacked damage stage awareness.

reaperrr committed:
Refactor RenderEditorOnly into RenderSpritesEditorOnly + WithSpriteBody

GraionDilach commented:
I'd say the YAML changes of the mines should be extracted out to a new PR with removing the RA1 mod defaults.

pchote commented on OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs:
```diff
@@ -81,7 +81,7 @@ public void PlayCustomAnimation(Actor self, string name, Action after = null)
 		public void PlayCustomAnimationRepeating(Actor self, string name)
 		{
-			DefaultAnimation.PlayThen(name,
+			DefaultAnimation.PlayThen(NormalizeSequence(self, name),
```
You normalize this one, but not the one below?

reaperrr committed:
Replace RenderBuilding where already possible

reaperrr commented:
Updated.

Mailaender commented:
Looks sensible. :+1: Editor only rendering still works. :white_check_mark:

pchote commented on OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs:
```diff
@@ -81,8 +81,8 @@ public void PlayCustomAnimation(Actor self, string name, Action after = null)
 		{
-			DefaultAnimation.PlayThen(name,
-				() => PlayCustomAnimationRepeating(self, name));
+			DefaultAnimation.PlayThen(NormalizeSequence(self, name),
```
Harvesting animations now appear to be double-normalized.  You should be able to adjust WithHarvestAnimation to account for these upstream checks (which I think are a good idea).

reaperrr committed:
Adapted WithHarvestAnimation to account for damage stages

reaperrr commented:
Updated.
For the record, I did test the harvester normalization in combination with damage stages, seems to work as intended.

reaperrr commented:
Updated, removed the `RenderSimple` commit. It will later return in its own PR, where `RenderSimple` will actually be removed from code as well. That PR will follow the complete removal of `RenderBuilding`.

reaperrr commented:
Updated.

penev92 commented:
Needs a rebase.

penev92 commented:
Looks and works OK.
:+1: when my comment on the last commit is addressed. Also when rebased, of course.

reaperrr commented:
Updated and rebased.

reaperrr commented:
Updated.

penev92 commented:
Thanks.
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/f307594884e7c8cd2552a8e857bade75ae05d0e6...c9903db9f8a964dc11006a3306022f18e3827045)
